diff --git a/src/wifi/model/block-ack-manager.cc b/src/wifi/model/block-ack-manager.cc
index 58dbcd5ba..4a1dfdf0a 100644
--- a/src/wifi/model/block-ack-manager.cc
+++ b/src/wifi/model/block-ack-manager.cc
@@ -885,6 +885,7 @@ BlockAckManager::InsertInRetryQueue (Ptr<WifiMacQueueItem> mpdu)
 
       it = m_retryPackets->PeekByTidAndAddress (tid, recipient, ++it);
     }
+  mpdu->GetHeader ().SetRetry ();
   m_retryPackets->Insert (it, mpdu);
 }
 
